packages/expo-sensors/src/Accelerometer.ts

```diff
@@ -1,4 +1,48 @@
+import type { Listener, Subscription } from './DeviceSensor';
+import DeviceSensor from './DeviceSensor';
 import ExponentAccelerometer from './ExponentAccelerometer';
-import ThreeAxisSensor from './ThreeAxisSensor';
 
-export default new ThreeAxisSensor(ExponentAccelerometer, 'accelerometerDidUpdate');
+/**
+ * Each of these keys represents the acceleration along that particular axis in g-force (measured in `g`s).
+ *
+ * A `g` is a unit of gravitational force equal to that exerted by the earth’s gravitational field (`9.81 m/s^2`).
+ */
+export type AccelerometerMeasurement = {
+  x: number;
+  y: number;
+  z: number;
+};
+
+export class AccelerometerSensor extends DeviceSensor<AccelerometerMeasurement> {
+  /**
+   * > **info** You should always check the sensor availability before attempting to use it.
+   *
+   * Returns whether the accelerometer is enabled on the device.
+   *
+   * On mobile web, you must first invoke `Accelerometer.requestPermissionsAsync()` in a user interaction (i.e. touch event) before you can use this module.
+   * If the `status` is not equal to `granted` then you should inform the end user that they may have to open settings.
+   *
+   * On **web** this starts a timer and waits to see if an event is fired. This should predict if the iOS device has the **device orientation** API disabled in
+   * **Settings > Safari > Motion & Orientation Access**. Some devices will also not fire if the site isn't hosted with **HTTPS** as `DeviceMotion` is now considered a secure API.
+   * There is no formal API for detecting the status of `DeviceMotion` so this API can sometimes be unreliable on web.
+   *
+   * @return A promise that resolves to a `boolean` denoting the availability of the accelerometer.
+   */
+  isAvailableAsync = async (): Promise<boolean> => {
```

Comment:
To be honest I always prefer this syntax 👇 because it's then part of the class prototype so one function is shared across all instances, instead of being set on each instance separately. Use assignments only when you need to capture `this` (React callbacks).
```suggestion
  async isAvailableAsync(): Promise<boolean> {
```

Reply:
By the way, wouldn't it be enough to just declare that this function exists (without implementation)?